src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs

```diff
@@ -152,5 +152,50 @@ public static partial class Activator
         }
 
         private static T CreateDefaultInstance<T>() where T : struct => default;
+
+        public static Func<object?> CreateFactory([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] Type type, bool nonPublic)
+        {
+            if (type is null)
+                throw new ArgumentNullException(nameof(type));
+
+            if (type.UnderlyingSystemType is not RuntimeType rt)
+                throw new ArgumentException(SR.Arg_MustBeType, nameof(type));
+
+            // First create the factory instance.
+
+            ActivationFactory factory = new ActivationFactory(rt);
+            if (!nonPublic && !factory.CtorIsPublic)
+            {
+                throw new MissingMethodException(SR.Format(SR.Arg_NoDefCTor, rt));
+            }
+
+            // Then create a delegate to factory.CreateInstance, closed over the "this" parameter.
+
+            return (Func<object?>)factory.CreateDelegate((RuntimeType)typeof(Func<object?>));
```

Comment:
Why can't this just call `new Func<object?>(factory.CreateInstance)` ? This looks slow and complicated for no good reason.